Framework/Kernel/inc/MantidKernel/case_insensitive_map.h

```diff
@@ -0,0 +1,31 @@
+#ifndef CASE_INSENSITIVE_MAP_H
```

Comment:
Not really. Wasn't sure if we used camel case for just classes and lower case underscore for stuff that isn't e.g. ```make_*.h``` or whether that was just due mimicking std lib support.